webapp-django/crashstats/home/static/home/js/home.js

```diff
@@ -0,0 +1,386 @@
+/*global MG: true, socorro: true */
+
+$(function () {
+    'use strict';
+
+    // A polyfill for `endsWith`
+    if (!String.prototype.endsWith) {
+        String.prototype.endsWith = function(searchString, position) {
+            var subjectString = this.toString();
+            if (typeof position !== 'number' || !isFinite(position) || Math.floor(position) !== position || position > subjectString.length) {
+                position = subjectString.length;
+            }
+            position -= searchString.length;
+            var lastIndex = subjectString.indexOf(searchString, position);
+            return lastIndex !== -1 && lastIndex === position;
+        };
+    }
+
+    var COLORS = ['#6a3d9a', '#e31a1c', '#008800', '#1f78b4'];
+
+    var container = $('#homepage-graph-container');
+    var superSearchApiUrl = container.data('supersearch-api');
+    var productBuildTypesApiUrl = container.data('product-build-types-api');
+    var adiApiUrl = container.data('adi-api');
+
+    var product = container.data('product');
+    var versions = container.data('versions');
+    var platforms = container.data('platforms');
+    var duration = container.data('duration');
+
+    var pageTitle = $('title').text();
+
+    // Versions ending in -b are "magic" versions that will be split into all
+    // of the beta versions starting with that same version number.
+    var betaVersions = [];
+    for (var i = versions.length - 1; i >= 0; i--) {
+        if (versions[i].endsWith('b')) {
+            betaVersions.push(versions[i]);
+        }
+    }
+
+    // Set default dates.
+    var endDate = new Date();
+    endDate.setDate(endDate.getDate() - 1);  // We start with yesterday.
+    var startDate = new Date();
+    startDate.setDate(endDate.getDate() - duration);
+
```

Comment:
I was allowed mess with the duration to some number that is NOT one of the duration options
<img width="1816" alt="screen shot 2016-02-22 at 4 20 18 pm" src="https://cloud.githubusercontent.com/assets/26739/13233140/49a7ec4a-d980-11e5-9960-e18999fabfa1.png">
Should we not guard against that?